zerver/tests/test_auth_backends.py

```diff
@@ -954,7 +954,26 @@ def test_log_into_subdomain_when_using_invite_link(self):
              'key': confirmation_key,
              'terms': True})
         self.assertEqual(result.status_code, 302)
-        self.assertEqual(sorted(self.get_streams('new@zulip.com', realm)), stream_names)
+        self.assertEqual(sorted(self.get_streams(self.nonreg_email('newuser'), realm)), stream_names)
+        self.assertEqual(self.nonreg_user('newuser').realm, realm)
+
+        # Now try registering to another realm using the invite link
```

Comment:
I have to confess I can't quite tell what this new test is meant to test. :-/ From the conversation that led to this PR, is it maybe someone trying to register on the wrong realm (not matching the invite they got) and getting rejected, or taken back to the right realm?

Reply:
I don't think this is the confusion, but fwiw, if someone getn a link a.zulip.com/join/12345 and they tamper it to b.zulip.com/join/12345, I think the right behavior is the same as the ConfirmationKeyException.DOES_NOT_EXIST path (i.e. I don't think we need to try to redirect them to the right domain).